### DISCUSSION TITLE
Ability to query Results webpart by Date

### DISCUSSION THREAD

**Discussion by Gaztab** [microsoft-search/pnp-modern-search#2180] [Q&A]
Hey,
I've not seen it, so I suppose this is a request. I want to enter a query to filter the Results by ModifiedOWSDATE:Today-30

**Describe alternatives you've considered**
If there was a way to apply static date filters to the results, that'd be a good alternative. 

Thank you,
Gareth

**Comment by wobba** [comment]
`ModifiedOWSDATE` is a string property. Use `LastmodifiedTime:{Today-30}`. Also see https://microsoft-search.github.io/pnp-modern-search/usage/search-results/tokens/#date-tokens and KQL date keywords a bit down the page at https://www.techmikael.com/2014/03/s15e01-kql-basics.html

**Reply by Gaztab** [reply]
Thanks for this. I am putting this into the Query Template like:
**Path:SiteURL and LastModifiedTime:{Today-30}** 
When I do this I am getting no results back even though there are items changed within 30days in this Site URL. I've gone through those links and wasn't able to see what else I should be doing. I've made sure the 'selected properties' have LastModifiedTime selected and Results Source ID / Scope is set to 'LocalSharePointResults'.

Can you see anything obvious I am missing?

**Reply by Gaztab** [reply]
Further - If I change the {Today-30} to (Today-30) I get an error "HTTP:500 CorrelationID". Not sure if this helps see what I am missing?

**Reply by Gaztab** [reply]
Thank you for this. I have now got this working. I used LastModifiedTimeForRetention>{today-30} to get it working. Thank you for your help.

Can you tell me on this. The results window only returns full version 1.0, 2.0, etc and never 0.1, 2.1, etc is there a way to change this to include all versions of documents? Thanks

**Reply by wobba** [reply]
You didn't specify that you wanted newer than 30 days ago, you had `:` which for a date is the same as `=` and means items from 30 days ago exactly 🙂. See https://docs.microsoft.com/en-us/sharepoint/dev/general-development/keyword-query-language-kql-syntax-reference#property-operators-that-are-supported-in-property-restrictions
